### PR TITLE
Remove callsign column, use slug for it.

### PR DESCRIPTION
**Pull request by mitsuhiko**
This starts the work for #2896

The idea I have right now is to just get this in and merged and deployed to us. This will remove the column anyways going forward.  Then I make this and the old migrations dummies.  The intermediary data of project and a few others will be bad but that does not cause any harm from a quick testing.  South only looks at the last frozen state.

@dcramer @matt: can someone verify the above idea?

### CODE REVIEW

**Comment by mattrobenolt** [comment]
That seems ok, would we be able to just remove the old migrations out of the repository entirely? Not sure if south would care about there being holes in migrations, but if we don't need people to run them at all, probably a bit better to just leave them out imo.

And for our case, it's probably not worth running this migration here, since we can just drop the column with a DDL.

**Comment by mitsuhiko** [comment]
@mattrobenolt i mostly just want the frozen data to be correct which is why this migration is necessary at least to exist (and be empty). I don't feel like fixing up all the frozen data for migrations in between.

**Comment by mattrobenolt** [comment]
Oh, yeah, that may be a pain. :rage1:

**Comment by mattrobenolt** [comment]
@mitsuhiko So, this is going to require two deploys and we need to split this up.

* We first need a deploy that doesn't use the new column. So everything in here, except the migration.
* A deploy that drops the column

Without this, we're going to be erroring in the interim as soon as the column is dropped.

**Comment by mitsuhiko** [comment]
Oh yeah. That's a good point. Should not have squashed those changes together.

**Comment by mitsuhiko** [comment]
This is two commits now.

File: src/sentry/api/endpoints/organization_slugs.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from rest_framework.response import Response
+
+from django.db import transaction
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.models import Project
+from django.core.validators import validate_slug, ValidationError
+
+
+class SlugsUpdateEndpoint(OrganizationEndpoint):
```
Comment (dcramer) [inline src/sentry/api/endpoints/organization_slugs.py]: lets avoid documenting this as I'd prefer we remove it and just use a "bulk project update" endpoint instead

**Comment by mitsuhiko** [comment]
Split into two PRs.

**Pull request closed without merging**